Feature/handle UI impact

Renad-Alalfy opened the pull request:
handle UI impact by refactoring UI classes and test cases for suspend logic

Renad-Alalfy committed:
enable UI to support suspend functions

Renad-Alalfy committed:
refactor unit tests to support suspend functions

yusufnasserdev commented on src/main/kotlin/org/damascus/Main.kt:
```diff
@@ -7,7 +7,7 @@ import org.damascus.ui.PlanMateConsoleUi
 import org.koin.java.KoinJavaComponent.getKoin
 
-fun main() {
+suspend fun main() {
```
Why did you choose to make main suspend-able? instead of using a Co-routine scope where you would need to call a suspend-able function?

Asmaa7071 committed:
Feature/refactor koin di to use annotations (#84)

* add koin annotation dependency

* add data source module

* add app module start koin

* update all classes to use koin annotation

* add named annotation

---------

Co-authored-by: Asmaa Karam <asmaakaram793@gmai.com>

elsabagh committed:
migrate test coverage and gh action to kover (#81)

* refactor git action by kover

* add exclude packages

* add KoverIgnore annotation class

* pump up kover plugin version to 0.9.1 from 0.6.1

Necessary changes were done, since difference between the two version was 3 years.

* add @KoverIgnore to ignore main function

* update kover xml report path in coverage.yml

---------

Co-authored-by: Yusuf Nasser <yusufnassereng@gmail.com>

Renad-Alalfy committed:
remove suspend from main and use CoroutineScope

Renad-Alalfy committed:
Merge branch 'feature/suspend-data-source' into feature/handle-ui-impact

# Conflicts:
#	build.gradle.kts

Renad-Alalfy committed:
rename scope to appScope

HarithBashar committed:
add authentication user test cases (#72)

* add authentication user test cases

* refactor code

* refactor code for authenticate user login test

---------

Co-authored-by: Yusuf Nasser <yusufnassereng@gmail.com>

HarithBashar committed:
add create mate test cases (#86)

YousifMurad committed:
Feature/state dashboard UI (#80)

* add TaskStateController.kt, TaskStateCli.kt and integrate it with ProjectDashboardCli.kt

* bind the ProjectDashboardCli.kt to real states repository instead of dummyStates

* remove commented outdated code

* refactor remove old AppModule and update the current AppModule

* refactor updateTaskStateUseCase method in ManageTaskStateUseCase

* remove unused files

* refactor TaskStateCli.kt

* refactor ProjectDashboardCli.kt to suit the new packaging and structure

* fix UI packaging

* split TaskStateCli.kt into separated classes files

* delete old unused TaskStateCli.kt and ProjectDashboardCli.kt

* modify classes for koin annotation by adding @Single and delete AppModule.kt

* add test cases for the taskState package

* refactor by extract headers and usageRow variable into functions

* use invoke operator instead of start()

* revert updateTaskState to the old logic

* remove unnecessary displayed info. of projectReferencesCount

* remove try and catch since the exception already handled in use case

* refactor UpdateTaskStateUi and its test

* remove unnecessary test case

HarithBashar committed:
add manage mate test cases (#88)

Renad-Alalfy committed:
refactor main function

Renad-Alalfy committed:
Merge remote-tracking branch 'origin/develop' into feature/handle-ui-impact

# Conflicts:
#	src/main/kotlin/org/damascus/Main.kt
#	src/main/kotlin/org/damascus/data/repo/AuditLogsRepositoryImpl.kt
#	src/main/kotlin/org/damascus/data/repo/TaskStateRepositoryImpl.kt

Renad-Alalfy committed:
merge develop branch into handle ui impact branch

Renad-Alalfy committed:
refactor main function
